model/flow/execution_receipt.go

```diff
@@ -129,3 +132,69 @@ func (g ExecutionReceiptGroupedList) GetGroup(groupID Identifier) ExecutionRecei
 func (g ExecutionReceiptGroupedList) NumberGroups() int {
 	return len(g)
 }
+
+/*******************************************************************************
+GROUPING for ExecutionReceiptMeta information:
+allows to split a list of receipt meta information by some property
+*******************************************************************************/
+
+// ExecutionReceiptMetaList is a slice of ExecutionResultMetas with the additional
+// functionality to group them by various properties
+type ExecutionReceiptMetaList []*ExecutionReceiptMeta
+
+// ExecutionReceiptMetaGroupedList is a partition of an ExecutionReceiptMetaList
+type ExecutionReceiptMetaGroupedList map[Identifier]ExecutionReceiptMetaList
+
+// ExecutionReceiptMetaGroupingFunction is a function that assigns an identifier to each receipt meta
+type ExecutionReceiptMetaGroupingFunction func(*ExecutionReceiptMeta) Identifier
+
+// GroupBy partitions the ExecutionReceiptMetaList. All receipts that are mapped
+// by the grouping function to the same identifier are placed in the same group.
+// Within each group, the order and multiplicity of the receipts is preserved.
+func (l ExecutionReceiptMetaList) GroupBy(grouper ExecutionReceiptMetaGroupingFunction) ExecutionReceiptMetaGroupedList {
+	groups := make(map[Identifier]ExecutionReceiptMetaList)
+	for _, rcpt := range l {
+		groupID := grouper(rcpt)
+		groups[groupID] = append(groups[groupID], rcpt)
+	}
+	return groups
+}
+
+// GroupByExecutorID partitions the ExecutionReceiptMetaList by the receipts' ExecutorIDs.
+// Within each group, the order and multiplicity of the receipts is preserved.
+func (l ExecutionReceiptMetaList) GroupByExecutorID() ExecutionReceiptMetaGroupedList {
+	grouper := func(receipt *ExecutionReceiptMeta) Identifier { return receipt.ExecutorID }
+	return l.GroupBy(grouper)
+}
+
+// GroupByResultID partitions the ExecutionReceiptMetaList by the receipts' Result IDs.
+// Within each group, the order and multiplicity of the receipts is preserved.
+func (l ExecutionReceiptMetaList) GroupByResultID() ExecutionReceiptMetaGroupedList {
+	grouper := func(receipt *ExecutionReceiptMeta) Identifier { return receipt.ResultID }
+	return l.GroupBy(grouper)
+}
+
+// Size returns the number of receipts in the list
+func (l ExecutionReceiptMetaList) Size() int {
+	return len(l)
+}
+
+// GetGroup returns the receipts that were mapped to the same identifier by the
+// grouping function. Returns an empty (nil) ExecutionReceiptMetaList if groupID does not exist.
+func (g ExecutionReceiptMetaGroupedList) GetGroup(groupID Identifier) ExecutionReceiptMetaList {
+	return g[groupID]
+}
+
+// NumberGroups returns the number of groups
+func (g ExecutionReceiptMetaGroupedList) NumberGroups() int {
+	return len(g)
+}
+
+// Lookup generates a map from ExecutionReceipt ID to ExecutionReceiptMeta
+func (l ExecutionReceiptMetaList) Lookup() map[Identifier]*ExecutionReceiptMeta {
+	receiptsByID := make(map[Identifier]*ExecutionReceiptMeta)
```

Comment:
can pre-allocate map here

Reply:
done in https://github.com/onflow/flow-go/pull/546/commits/282e833f6c0e7903846076da949f7dbf625f6e9e